### PR TITLE
[ISSUE-252] refactor(cashout): use ValidatedInput for account number

### PR DESCRIPTION
**Pull request by jjramirezn**
Added the ValidatedInput component to check if the account number is valid and show feedback to the user before they try to proceed

### CODE REVIEW

File: src/components/Cashout/Components/Initial.view.tsx [hunk 1]
```diff
@@ -281,35 +262,30 @@ export const InitialCashoutView = ({
...
                                                     onClick={() => {
                                                         if (!xchainAllowed) return
-                                                        if (selectedBankAccount === account.account_identifier) {
-                                                            setSelectedBankAccount(undefined)
-                                                        } else {
-                                                            setSelectedBankAccount(account.account_identifier)
-                                                            setActiveInput('selectedBankAccount')
-                                                        }
+                                                        setAccountNumber(account.account_identifier)
                                                     }}
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/components/Cashout/Components/Initial.view.tsx]: those account should not generate error states, and if so the ValidatedInput component would clear any error message

File: src/components/Cashout/Components/Initial.view.tsx [hunk 2]
```diff
@@ -61,6 +61,9 @@ export const InitialCashoutView = ({
     const [_tokenValue, _setTokenValue] = useState<string | undefined>(
         inputDenomination === 'TOKEN' ? tokenValue : usdValue
     )
+    const [accountNumber, setAccountNumber] = useState<string>('')
```
Comment (panosfilianos) [inline src/components/Cashout/Components/Initial.view.tsx]: issue (blocking): what is this accountNumber? can we comment? (i.e. bankAccountNumber)
the same for the additions below

File: src/components/Cashout/Components/Initial.view.tsx [hunk 3]
```diff
@@ -281,35 +262,30 @@ export const InitialCashoutView = ({
...
                                                     </div>
                                                     <div className="flex w-6 justify-center">
-                                                        {selectedBankAccount === account.account_identifier && (
+                                                        {accountNumber === account.account_identifier && (
                                                             <button
                                                                 className="text-lg text-black"
                                                                 onClick={(e) => {
                                                                     e.stopPropagation()
-                                                                    setSelectedBankAccount(undefined)
+                                                                    setAccountNumber('')
```
Comment (panosfilianos) [inline src/components/Cashout/Components/Initial.view.tsx]: thought (non-blocking): imo it's better to have this typed as `<string | undefined>` bc we can run ifs like `if(accountNumber)` more cleanly. An empty string is falsy, but it just feels cleaner to me.

**Review by panosfilianos — CHANGES_REQUESTED** [review]
I think a small naming change would be helpful here. This comes from me playing around w/ wallets currently and seeing there can be many hooks/modules w/ same naming (i.e. account) and it becomes harder/ slower to disambiguate.

wdyt?

File: src/components/Cashout/Components/Initial.view.tsx (hunk 2 above)
Comment (jjramirezn) [reply src/components/Cashout/Components/Initial.view.tsx]: good idea, implementing

File: src/components/Cashout/Components/Initial.view.tsx (hunk 3 above)
Comment (jjramirezn) [reply src/components/Cashout/Components/Initial.view.tsx]: In the end [inputs need a defined value](https://react.dev/reference/react-dom/components/input#im-getting-an-error-a-component-is-changing-an-uncontrolled-input-to-be-controlled) I prefer to have the state of bankAccountNumber and the value on the input to always be the same (as opposed to the state being undefined and the input be empty string)

File: src/components/Cashout/Components/Initial.view.tsx [hunk 4]
```diff
@@ -86,36 +85,32 @@ export const InitialCashoutView = ({
...
         )
-    }, [_tokenValue, selectedBankAccount, newBankAccount, xchainAllowed, isBelowMinLimit, isExceedingMaxLimit])
+    }, [
+        _tokenValue,
+        isValidBankAccountNumber,
+        isValidatingBankAccountNumber,
+        xchainAllowed,
+        isBelowMinLimit,
+        isExceedingMaxLimit,
+    ])
```
Comment (nezz0746) [inline src/components/Cashout/Components/Initial.view.tsx:101]: nit: booleans like these variables usually don't need useMemo, you're not getting any performance improvement. Also it includes potential bugs if you're missing a variable in the dependency array the the compute is broken

**Review by nezz0746 — APPROVED** [review]

File: src/components/Cashout/Components/Initial.view.tsx (hunk 3 above)
Comment (panosfilianos) [reply src/components/Cashout/Components/Initial.view.tsx]: agreed

**Review by panosfilianos — APPROVED** [review]

**Pull request merged**